sdk/apis/kubermatic/v1/cluster.go

```diff
@@ -677,6 +677,10 @@ type ClusterStatus struct {
 
 	// ResourceUsage shows the current usage of resources for the cluster.
 	ResourceUsage *ResourceDetails `json:"resourceUsage,omitempty"`
+
+	// ProviderStatus holds the status information for all cluster providers.
+	// +optional
+	ProviderStatus *ClusterProviderStatus `json:"providerStatus,omitempty"`
 }
```

Comment:
Just `Provider` or `CloudProvider` is fine, the name of struct is `Status` so no need to append that in the fields.